app/Domain/Integrations/Controllers/IntegrationController.php

```diff
@@ -142,41 +134,61 @@ public function destroyUrl(Request $request, string $id, string $urlId): Redirec
         );
     }
 
-    public function update(UpdateIntegrationRequest $request, string $id): RedirectResponse
+    public function updateIntegration(UpdateIntegrationRequest $request, string $id): void
     {
         $currentIntegration = $this->integrationRepository->getById(Uuid::fromString($id));
 
-        $updatedIntegration = UpdateIntegrationMapper::map($request, $currentIntegration);
+        if($currentIntegration->name !== $request->get('integrationName') || $currentIntegration->description !== $request->get('description')) {
+            $updatedIntegration = UpdateIntegrationMapper::map($request, $currentIntegration);
 
-        $this->integrationRepository->update($updatedIntegration);
-
-        return Redirect::route(
-            TranslatedRoute::getTranslatedRouteName($request, 'integrations.show'),
-            [
-                'id' => $id,
-            ]
-        );
+            $this->integrationRepository->update($updatedIntegration);
+        }
     }
 
-    public function updateUrls(UpdateIntegrationUrlsRequest $request, string $id): RedirectResponse
+    public function update(UpdateIntegrationUrlsRequest $request, string $id): RedirectResponse
     {
+        $integrationRequest = ['integrationName' => $request->input('integrationName'), 'description' => $request->input('description')];
+
+        $integrationUpdateRequest = new UpdateIntegrationRequest($integrationRequest);
+
+        $this->updateIntegration($integrationUpdateRequest, $id);
+
         $ids = array_map(
             fn ($url) => Uuid::fromString($url['id']),
             array_filter(
                 [
-                    $request->input('loginUrl'),
+                    ...($request->input('loginUrls') ?? []),
                     ...($request->input('callbackUrls') ?? []),
                     ...($request->input('logoutUrls') ?? []),
                 ],
                 fn ($val) => $val !== null
             )
         );
 
-        $currentUrls = $this->integrationUrlRepository->getByIds($ids);
+        if (count($ids) > 0) {
+            $currentUrls = $this->integrationUrlRepository->getByIds($ids);
 
-        $updatedUrls = UpdateIntegrationUrlsMapper::map($request, $currentUrls);
+            $updatedUrls = UpdateIntegrationUrlsMapper::map($request, $currentUrls);
 
-        $this->integrationUrlRepository->updateUrls($updatedUrls);
+            $this->integrationUrlRepository->updateUrls($updatedUrls);
+        }
+
+        $newUrls =
+            array_filter(
+                [
+                    ...($request->input('newIntegrationUrls') ?? []),
+                ],
```

Comment:
```suggestion
                ($request->input('newIntegrationUrls') ?? []),
```

No need to create a new array